src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawling_context.py

```diff
@@ -1,26 +1,5 @@
-from __future__ import annotations
+from bs4 import BeautifulSoup
 
-from dataclasses import dataclass
-from typing import TYPE_CHECKING
+from crawlee.http_crawler import ParsedHttpCrawlingContext
 
-from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction
-from crawlee._utils.docs import docs_group
-from crawlee.http_crawler import HttpCrawlingResult
-
-if TYPE_CHECKING:
-    from bs4 import BeautifulSoup
-
-
-@dataclass(frozen=True)
-@docs_group('Data structures')
-class BeautifulSoupCrawlingContext(HttpCrawlingResult, BasicCrawlingContext):
-    """The crawling context used by the `BeautifulSoupCrawler`.
-
-    It provides access to key objects as well as utility functions for handling crawling tasks.
-    """
-
-    soup: BeautifulSoup
-    """The `BeautifulSoup` object for the current page."""
-
-    enqueue_links: EnqueueLinksFunction
-    """The BeautifulSoup `EnqueueLinksFunction` implementation."""
+BeautifulSoupCrawlingContext = ParsedHttpCrawlingContext[BeautifulSoup]
```

Comment:
A `soup` property would be helpful for BC.

Reply:
It has it from ParsedHttpCrawlingContext. It is just an alias for ParsedHttpCrawlingContext[BeautifulSoup]

I tried creating standalone class BeautifulSoupCrawlingContext(ParsedHttpCrawlingContext[BeautifulSoup]) with it's own property soup, but that is no longer accepted by Mypy static check as it is children class of ParsedHttpCrawlingContext[BeautifulSoup] and that parent class is expected by mypy to have soup property. Unfortunately the only way how I figured out to work around it was to define both of those soup (from BScrawler) and selector (from ParselCrawler) properties on ParsedHttpCrawlingContext and do some runtime checks. Not something I would be proud of, but last resort. This solution is statically safe and backwards compatible. The only "statically unsafe" action is, when for example someone using ParselCrawler will try to access context.soup -> kind of a missuse edge case that will throw exception that exactly describes what should be done instead.

Reply:
Discussed in person